### ISSUE TITLE
Add support for bzlmod

### ISSUE THREAD

**Issue by UebelAndre** [bazelbuild/rules_rust#1493]
It would be nice if `rules_rust` supported [bzlmod](https://docs.bazel.build/versions/main/bzlmod.html) workflows. Hopefully someone familiar with that can enable that for this project.

**Comment by cameron-martin** [comment]
I don't know that much about bzlmod, but have started having a look at this. Initially I would like to simply replace `rules_rust_dependencies`. However, I'm having some issues. Details in #1528.

**Comment by hobofan** [comment]
I think it makes sense to reopen this? While the work that has been merge, from what I can tell there is still quite a bit of work that needs to happen to "support bzlmod" on a level that it can actually be adopted.

**Comment by katre** [comment]
Are loading crates via bzlmod supported in the current effort?

**Comment by opicaud** [comment]
Hey, 
Please check #2021

**Comment by csmulhern** [comment]
Bazel 7.0 is releasing later this year, with bzlmod becoming on by default. Is anyone working to prioritize this work to align with the release of Bazel 7.0? Looks like there has been some work done in #2021.

**Comment by UebelAndre** [comment]
It unfortunately seems no maintainer has any experience with bzlmod at all. I've asked other members of the Bazel org to help do reviews but I unfortunately haven't seen any support. The Bzlmod changes could probably be merged as long as there's no change to existing code. For anything that needs to change, the more granular the pull-reqeusts can be the easier the rollout will be. My primary concern is that bzlmod just increases the maintenance burden by introducing a second way to to fetch dependencies. So it'd also be helpful if the changes were setup to maintain a single source of truth.

**Comment by scentini** [comment]
I have scheduled some time with Bzlmod experts later this week to peer review this PR with the hope that we can move this PR forward and gain some expertise in the process.

**Comment by csmulhern** [comment]
Makes sense. Thanks for the update.

I would hope long term, Bzlmod becomes the primarily supported way to fetch dependencies, though that's probably quite far in the future.

Shot in the dark, but I have tried to solicit help in this discussion thread: https://github.com/bazelbuild/bazel/discussions/18329.

**Comment by csmulhern** [comment]
> I have scheduled some time with Bzlmod experts later this week to peer review this PR with the hope that we can move this PR forward and gain some expertise in the process.

That sounds great, thanks @scentini. I am happy to dedicate some bandwidth in the coming month to help with contributions (though I also have no prior expertise).

**Comment by UebelAndre** [comment]
> I have scheduled some time with Bzlmod experts later this week to peer review this PR with the hope that we can move this PR forward and gain some expertise in the process.

@scentini any updates from your meeting?

**Comment by chrislovecnm** [comment]
Hi all! Any updates? I have some bandwidth to help.

**Comment by chrislovecnm** [comment]
This needs some tlc https://bazelbuild.github.io/rules_rust/

**Comment by cameron-martin** [comment]
Should this be closed now? Bzlmod seems to be quite well supported by rules_rust.

**Comment by marvin-hansen** [comment]
Yes!

**Comment by illicitonion** [comment]
Yeah, I'd say we have bzlmod support - there are a few corners that may be missing or could be improved, but those can be tracked as standalone issues.
